src/main/kotlin/lotto/LottoApplication.kt

```diff
@@ -1,22 +1,33 @@
 package lotto
 
-import lotto.domain.BonusManualGenerateStrategy
-import lotto.domain.LottoAutoGenerateStrategy
 import lotto.domain.LottoMachine
-import lotto.domain.LottoManualGenerateStrategy
+import lotto.domain.input.ConsoleInput
+import lotto.domain.strategy.bonus.BonusManualGenerateStrategy
+import lotto.domain.strategy.lotto.LottoAutoGenerateStrategy
+import lotto.domain.strategy.lotto.LottoManualGenerateStrategy
 import lotto.view.InputView
 import lotto.view.OutputView
 
 fun main() {
-    val money = InputView.read()
-    val lottoMachine = LottoMachine(
-        money = money,
-        lottoGenerateStrategy = LottoAutoGenerateStrategy(),
-        winnerLottoGenerateStrategy = LottoManualGenerateStrategy(),
-        bonusGenerateStrategy = BonusManualGenerateStrategy()
-    )
-    OutputView.printPurchaseTicketResult(lottoMachine.lottoTickets)
-    OutputView.printWinnerTicket()
-    val lottoSummary = lottoMachine.execute()
-    OutputView.printSummary(lottoSummary)
+    runCatching {
+        val money = InputView.readMoney()
+        val manualTicketCount = InputView.readManualTicketCount()
+        OutputView.printManualTicketEnter()
+        val lottoMachine = LottoMachine(
+            money = money,
+            manualTicketCount = manualTicketCount,
+            lottoGenerateStrategies = listOf(LottoAutoGenerateStrategy(), LottoManualGenerateStrategy(ConsoleInput())),
+            winnerLottoGenerateStrategy = LottoManualGenerateStrategy(ConsoleInput()),
+            bonusGenerateStrategy = BonusManualGenerateStrategy(ConsoleInput())
+        )
+        OutputView.printPurchaseTicketResult(lottoMachine.lottoTickets)
+        OutputView.printWinnerTicket()
+        val lottoSummary = lottoMachine.execute()
+        OutputView.printSummary(lottoSummary)
+    }.onFailure {
+        when (it) {
+            is IllegalArgumentException -> throw it
+            else -> null
```

Comment:
```suggestion
    }.onFailure {
        when (it) {
            is IllegalArgumentException -> throw it
```

onFailure는 반환 타입이 Unit이에요. 의미 없는 두번째 else 구문은 제거하는게 좋을 것 같아요.

```kotlin
inline fun <T> Result<T>.onFailure(
    action: (exception: Throwable) -> Unit
): Result<T>
```